2006-05-02  Vesa Jaaskelainen  <chaac@nic.fi>

       * stage2/shared.h (vbe_mode): Back ported aligment fix from GRUB 2
       to GRUB Legacy.  Problem reported by Gerardo Richarte.

taken from upstream.

Index: stage2/shared.h
===================================================================
RCS file: /cvsroot/grub/grub/stage2/shared.h,v
retrieving revision 1.99
retrieving revision 1.100
diff -u -r1.99 -r1.100
--- foo/stage2/shared.h	20 Jun 2004 13:48:47 -0000	1.99
+++ foo/stage2/shared.h	2 May 2006 20:46:24 -0000	1.100
@@ -499,7 +499,11 @@
   unsigned char linear_reserved_field_position;
   unsigned long max_pixel_clock;
 
-  unsigned char reserved3[189];
+  /* Reserved field to make structure to be 256 bytes long, VESA BIOS 
+     Extension 3.0 Specification says to reserve 189 bytes here but 
+     that doesn't make structure to be 256 bytes.  So additional one is 
+     added here.  */
+  unsigned char reserved3[189 + 1];
 } __attribute__ ((packed));
 
 
